lib/plugins/webpagetest/analyzer.js

```diff
@@ -93,10 +94,12 @@ module.exports = {
 
           }
         })
-        return Promise.all(promises).then((result) => {
+
+        // set a low concurrency to not overload the WebPageTest server
+        return Promise.map(promises, function(){}, {concurrency: 2}).then(() => {
```

Comment:
Sorry, but this will not have any effect. To quote the docs: 'The concurrency limit applies to Promises returned by the mapper function'. I.e. it waits on the promises returned by the mapper function, and keeps track of the number of currently unresolved promises. However this mapper function returns undefined, so the concurrency option will have no effect. Additionally since the async calls have already been made (when pushing the wptClient.get…Async result promises to the promises array), this doesn't affect the number of concurrent http calls.

To control this we need a queue, concurrent-queue as used for QueueHandler should work.